### ISSUE TITLE
VSCODE not recognising relative file paths in the terminal output

### ISSUE THREAD

**Issue by TRACTOR-MAN** [microsoft/vscode#75841]
- VSCode Version: 1.35.1
- OS Version: windows 7

Steps to Reproduce:

1. Create sample project folder
2. Create two folders inside the project folder DIR1 and DIR2
3. Create a workspace inside one of the folders created in DIR1
4. Create a file Test.txt in the other folder created in DIR2
5 Add the project folder (1) into the workspace.
6 Create a tasks.json file in a .vscode folder inside the directory created in (1) containing:
{
    "version": "2.0.0",
    "tasks":
    [
        {
            "label": "BUILD ",
            "type": "shell",
            "command": "echo DIR_2\\Test.txt",
            "options": {
                "cwd": "${workspaceFolder}" /* The workspace folder is the same folder as the workspace file */
            },
            "group": "build",
            "problemMatcher": {"base": "$gcc", "fileLocation": ["relative", "${workspaceRoot}"]
        }
    ]
}

Does this issue occur when all extensions are disabled?: Yes

This is a simplified project to show the issue.  I am currently working on two C projects, and in one the links work.  In the other the links don't work no matter what I try.  There appears to be some issue with how vscode is parsing the links.

**Comment by alexr00** [comment]
duplicate of #69403

**Comment by TRACTOR-MAN** [comment]
Thanks, sorry that it's a duplicate. Just found it odd that it works for one project but not for others, which used to the best of my knowledge identical vscode setups.

**Comment by TRACTOR-MAN** [comment]
Hi I have been doing some more testing over the weekend with the insiders version 1.36.0.

I have found that the issue seems to be fixed on both linux and windows.

However, I came back to work and found that the issue was still present on my windows 7 machine.  I did some more digging and realized that the link detection algorithm does not work when there are any spaces in the file path from C: to my workspace folder.  Basically if there are no spaces it works fine, if there are any at all it does not.  Perhaps this could be an improvement for the future.
